app/controllers/application_controller.rb

```diff
@@ -1,2 +1,11 @@
 class ApplicationController < ActionController::Base
+  before_action :set_locale
+  private
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
```

Comment:
```suggestion
  before_action :set_locale
  
  private
  
  def set_locale
    I18n.locale = params[:locale] || I18n.default_locale
  end
```
Giữa các thành phần khác nhau thì thêm 1 line trống để clear code hơn

Reply:
dạ a